### ISSUE TITLE
CA1508 incorrect string semantics in VB.NET

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn-analyzers#1922]
#### Analyzer package

Microsoft.CodeQuality.Analyzers.Exp

#### Package Version

v2.6.3-beta1.18611.3

#### Diagnostic ID

CA1508 - Remove or refactor conditions to avoid dead code

#### Repro steps

CA1508 triggers on this code with the inverted logic:

```vb.net
Dim foo As String = Nothing
If foo = "" Then
    foo = "Foo"
End If
```

The current diagnostic states:

> 'foo = ""' is always 'False'

VB has different semantics around `= ""` for strings:

```vb.net
Dim s As String = Nothing
Console.WriteLine(s = "") ' prints "True"
```

#### Expected behavior

Diagnostic indicates expression is alwasy `true`.

#### Actual behavior

Diagnostic indicates expression is alwasy `false`.
